### PR TITLE
Add ReRequestCheckRun

### PR DESCRIPTION
**Pull request by leonpham0**
Closes out https://github.com/google/go-github/issues/2357

Adds support for https://docs.github.com/en/rest/checks/runs#rerequest-a-check-run

### CODE REVIEW

File: github/checks.go
Comment (gmlewis) [inline github/checks.go:324]:
```suggestion
	return s.client.Do(ctx, req, nil)
```

**Review by gmlewis — CHANGES_REQUESTED** [review]
Thank you, @leonpham0 !
Just one minor tweak, please, then we should be ready for a second LGTM+Approval before merging.

**Review by gmlewis — APPROVED** [review]
Thank you, @leonpham0 !
LGTM.

Awaiting second LGTM+Approval from any other contributor to this repo before merging.

**Review by raynigon — APPROVED** [review]
LGTM

**Comment by gmlewis** [comment]
> LGTM

Thank you, @raynigon !
Merging.

**Pull request merged**
